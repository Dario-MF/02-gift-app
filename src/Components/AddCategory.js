import React, { useState } from 'react'
import Proptypes from 'prop-types'

const AddCategory = ({ setCategories }) => {
    const [category, setCategory] = useState('Ingrese categoria')

    const handleChange = e => {

        setCategory(e.target.value)
    }
    const handleSubmit = e => {
        e.preventDefault()
        if (category.trim().length > 2) {
            setCategories(categories => [category, ...categories])
            setCategory('')
        }
    }
    const handleFocus = e => {
        e.target.value = ''
    }

    return (
        <form onSubmit={handleSubmit}>
            <h2>{category}</h2>
            <input
                type='text'
                value={category}
                onChange={handleChange}
                onFocus={handleFocus}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: Proptypes.func.isRequired
}

export default AddCategory