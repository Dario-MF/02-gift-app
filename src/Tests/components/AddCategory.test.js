import { shallow } from 'enzyme';
import React from 'react'
import AddCategory from '../../Components/AddCategory';

describe('Test for <AddCategory />', () => {
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);
    beforeEach(()=> {
        jest.clearAllMocks()
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    })

    test('test snapShot', () => {
       
        expect(wrapper).toMatchSnapshot()
    })

    test('test for capture imput.value', () => {
        const input = wrapper.find('input');
        const value = 'Hola mundo'
        input.simulate('change',{target:{ value: value }})

    })

    test('No debe de llamar a setCategories si el imput no pasa la condicion', () => {
        wrapper.find('form').simulate('submit', {preventDefault(){}})

        expect(setCategories).not.toHaveBeenCalled()
    })
    

    
    
})
