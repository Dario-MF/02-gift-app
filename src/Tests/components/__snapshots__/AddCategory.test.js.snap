// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for <AddCategory /> test snapShot 1`] = `
<form
  onSubmit={[Function]}
>
  <h2 />
  <input
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value=""
  />
</form>
`;
